### ISSUE TITLE
IPR has to be generated before user selects ILP as a payment method

### ISSUE THREAD

**Issue by michielbdejong** [michielbdejong/webpayments-demo#8]
In the current flow, the IPR is generated as soon as the user clicks 'Buy', so if they then still pay with creditcard, the IPR is wasted.

**Comment by michielbdejong** [comment]
would be good to only pass
* a shopping-cart-id so the shop knows which session you're paying for
* the shop's ILP address

and then wait until the payment app for generating the condition.

even better, additionally pass the shop's public key, and use Sender-Condition-to-Public-Key flow ('ekp')

**Comment by adrianhopebailie** [comment]
The new proposal for SPSP would mean the merchant could simply pass a URL in the request. The payee would perform a GET on that URL and use the shared secret to generate the condition.

See: https://github.com/interledger/rfcs/issues/152

**Comment by michielbdejong** [comment]
Yes, a first step towards this is if I pass [this URL](https://github.com/michielbdejong/webpayments-demo/blob/gh-pages/merchants/clothing/index.html#L92) instead of already doing the xhr and passing the result of it.
